tests/src/system/basic/WskActionTests.scala

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package system.basic
+
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+
+import common.JsHelpers
+import common.TestHelpers
+import common.TestUtils
+import common.Wsk
+import common.WskProps
+import common.WskTestHelpers
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import spray.json.JsObject
+import spray.json.pimpAny
+
+@RunWith(classOf[JUnitRunner])
+class WskActionTests
+    extends TestHelpers
+    with WskTestHelpers
+    with JsHelpers {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk(usePythonCLI = false)
+
+    val testString = "this is a test"
+    val testResult = JsObject("count" -> testString.split(" ").length.toJson)
+
+    behavior of "Whisk actions"
+
+    it should "invoke an action with a space in the name" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "hello Async"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/helloAsync.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status shouldBe "success"
+                    activation.response.result shouldBe Some(testResult)
+                    activation.logs.get.mkString(" ") should include(testString)
+            }
+    }
+
+    it should "pass parameters bound on creation-time to the action" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "printParams"
+            val params = Map(
+                "param1" -> "test1",
+                "param2" -> "test2")
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) =>
+                    action.create(
+                        name,
+                        Some(TestUtils.getCatalogFilename("samples/printParams.js")),
```

Comment:
Since I am refactoring all the test cases to remove the catalog, could you change getCatalogFilename into getTestActionFilename? If the script is not available under tests/dat/actions, please add it.